File: accommodation_service/src/main/java/huy/project/accommodation_service/infrastructure/repository/model/ImageModel.java
package huy.project.accommodation_service.infrastructure.repository.model;

import jakarta.persistence.*;
import lombok.*;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
@Entity
@Table(name = "images")
public class ImageModel extends AuditTable {
    @Id
    private Long id;

    @Column(name = "url")
    private String url;

    @Column(name = "entity_id")
    private Long entityId;

    @Column(name = "entity_type")
    private String entityType;

    @Column(name = "is_primary")
    private Boolean isPrimary;
}

File: location_service/LocationService.Infrastructure/Repository/Model/ImageModel.cs
namespace LocationService.Infrastructure.Repository.Model
{
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;

    [Table("images")]
    public class ImageModel : AuditModel
    {
        [Key]
        [Column("id")]
        public long Id { get; set; }

        [Column("entity_id")]
        public long EntityId { get; set; }
        
        [Column("url")]
        public string Url { get; set; }

        [Column("entity_type")]
        public string EntityType { get; set; }

        [Column("is_primary")]
        public bool IsPrimary { get; set; }
    }
}